globus_action_provider_tools/authentication.py

```diff
@@ -190,11 +192,10 @@ def get_dependent_tokens(self, bypass_cache_lookup=False) -> OAuthTokenResponse:
         resp = self.auth_client.oauth2_get_dependent_tokens(
             self.bearer_token, additional_params={"access_type": "offline"}
         )
-        if resp is not None and dependent_tokens_cache_id is not None:
-            log.info(
-                f"Caching dependent token response for token ***{self.sanitized_token}"
-            )
-            AuthState.dependent_tokens_cache[dependent_tokens_cache_id] = resp
+        log.info(
+            f"Caching dependent token response for token ***{self.sanitized_token}"
```

Comment:
Would it be worthwhile to incorporate the `"***"` prefix into the `sanitized_token` value directly?

Reply:
Maybe? But I'd rather just get rid of that suffix-of-token-as-"sanitized" stuff.

A *very* early version of the SDK did this for its debug-level logging, and it seems to have accidentally promulgated the idea that a suffix of the token is a good value to use in logs. A hash is probably better for multiple reasons, and we could maybe even use the same hashing algo as Auth so that it matches the `at_hash` claim of the `id_token`, which might give us some nice symmetry.